### ISSUE TITLE
Path not refeshed after switch to powerline-shell prompt

### ISSUE THREAD

**Issue by a-dekker** [b-ryan/powerline-shell#405]
Although I really like this prompt, I still prefer bash-git-prompt (https://github.com/magicmonty/bash-git-prompt) when working in git repos.
So I configured bash-git-prompt to be active only in git repos (GIT_PROMPT_ONLY_IN_REPO=1).
Unfortunately, powerline-shell does not update the current path then leaving a git repo. It shows the old path at first. After some action on the prompt, the path updates itself.

_Example_ (entering git repo (activating bash-git-prompt), then cd back, followed by enter to update the old path):
```
 user  servername  ~  $  cd /path/to/git/repo/
✔  /path/to/git/repo/ [master|✚ 23]
10:52 $ cd
 user  servername  path  to  git  repo  $ 
 user  servername  ~  $ 
```

Without powerline-shell, the (default) prompt is updated correctly at once after leaving a git repo.

**Comment by b-ryan** [comment]
This is a unique use-case. The issue appears to be caused because the bash-git-prompt code runs after the powerline-shell code. I think what you want to do is change up your `PROMPT_COMMAND` so that powerline-shell runs after bash-git-prompt, but also change the `_update_ps1` function so it does not run when in a git directory. Change the last lines of the above to this:

```
function _update_ps1() {
    local repo=$(git rev-parse --show-toplevel 2> /dev/null)
    if [[ ! -e "$repo" ]] && [[ "$GIT_PROMPT_ONLY_IN_REPO" = 1 ]]; then
        PS1="$(powerline-shell $?)"
    fi
}

if [[ $TERM != linux && ! $PROMPT_COMMAND =~ _update_ps1 ]]; then
    PROMPT_COMMAND="$PROMPT_COMMAND; _update_ps1;"
fi
```

I adapted code from [here](https://github.com/magicmonty/bash-git-prompt/blob/19c8def6a90d72f2a99d781752271126b1b9ff66/gitprompt.sh#L323-L328) for the `_updated_ps1` function. I did not test all this, but I think it should do the trick.
